### PR TITLE
Fix Dependabot PRs

### PR DESCRIPTION
**Pull request by chatton**
Dependabot PRs are supplied with a **read only** GitHub token. This means that secrets are not passed to our tests, which means our tests cannot run.

This PR updates the condition for which our tests run which should allow the use of the `safe-to-test` label on dependabot PRs.

### CODE REVIEW

File: .action_templates/e2e-pr-template.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: Run E2E
 jobs:
   - template: setup
-    if: github.event.pull_request.head.repo.full_name == 'mongodb/mongodb-kubernetes-operator'
+    if: github.event.pull_request.head.repo.full_name == github.repository && github.actor != 'dependabot[bot]'
```
Comment (chatton) [inline .action_templates/e2e-pr-template.yaml:4]: regular e2e tests as normal for non depenbot PRs.

File: .action_templates/e2e-fork-template.yaml [hunk 2]
```diff
@@ -1,7 +1,8 @@
 name: Run E2E Fork
 jobs:
   - template: setup
-    if: github.event.pull_request.head.repo.full_name != 'mongodb/mongodb-kubernetes-operator' && contains(github.event.pull_request.labels.*.name, 'safe-to-test')
+    # dependabot gets a read only github token, and so must use pull_request_target instead of pull_request.
+    if: github.actor == 'dependabot[bot]' || (github.event.pull_request.head.repo.full_name != github.repository && contains(github.event.pull_request.labels.*.name, 'safe-to-test'))
```
Comment (chatton) [inline .action_templates/e2e-fork-template.yaml:5]: run tests for dependabot always, same logic with labelling for PRs from forks.

File: requirements.txt [hunk 3]
```diff
@@ -1,5 +1,5 @@
 git+https://github.com/mongodb/sonar@0.0.11#egg=sonar
-git+https://github.com/chatton/GithubActionTemplates@v0.0.2
+git+https://github.com/chatton/GithubActionTemplates@v0.0.4
```
Comment (chatton) [inline requirements.txt:2]: fixed an issue with incorrect default directory

File: .github/workflows/e2e.yml [hunk 4]
```diff
@@ -37,7 +37,8 @@ jobs:
         - pipeline-argument: agent-ubi
         - pipeline-argument: agent-ubuntu
         - pipeline-argument: e2e
-    if: github.event.pull_request.head.repo.full_name == 'mongodb/mongodb-kubernetes-operator'
+    if: github.ref == 'refs/heads/master' || (github.event.pull_request.head.repo.full_name
```
Comment (chatton) [inline .github/workflows/e2e.yml:40]: @irajdeep this should fix the issue with tests not working from master.

**Review by irajdeep — APPROVED** [review]

**Pull request merged**